store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -1110,7 +1110,7 @@ private boolean isTheBatchFull(int sizePy, int maxMsgNums, int bufferTotal, int
             return false;
         }
 
-        if ((messageTotal + 1) >= maxMsgNums) {
```

Comment:
I will suggest  replace all the condition of `messageTotal +1 > xxx` into 
    
    messageTotal >= xxx